File: src/test/groovy/com/benjaminsproule/swagger/gradleplugin/test/jaxrs/ExtendedTestResourceWithClassAnnotation.java
package com.benjaminsproule.swagger.gradleplugin.test.jaxrs;

import javax.ws.rs.GET;
import javax.ws.rs.Path;

import io.swagger.annotations.ApiOperation;

public class ExtendedTestResourceWithClassAnnotation extends TestResourceWithClassAnnotation {

    @ApiOperation("An overriden operation description")
    @Path("/overriden")
    @GET
    @Override
    public String overriden() {
        return "";
    }
}

File: src/test/scala/com/benjaminsproule/swagger/gradleplugin/test/scala/jaxrs/ExtendedTestResourceWithClassAnnotation.scala
package com.benjaminsproule.swagger.gradleplugin.test.scala.jaxrs

import io.swagger.annotations.ApiOperation
import javax.ws.rs.{GET, Path}

class ExtendedTestResourceWithClassAnnotation extends TestResourceWithClassAnnotation {

    @ApiOperation("An overriden operation description")
    @Path("/overriden")
    @GET
    @Override
    override def overriden(): String = {
        ""
    }

    @Path("/overridenWithoutDescription")
    @GET
    override def overridenWithoutDescription(): String = {
        ""
    }
}
